File: solutions/go/difference-of-squares/1/difference_of_squares.go
package diffsquares

func sqr(n int) int {
	return n * n
}

func SquareOfSum(n int) int {
	sum := 0
	for i := 0; i <= n; i++ {
		sum += i
	}

	return sqr(sum)
}

func SumOfSquares(n int) int {
	if n == 1 {
		return 1
	} else {
		return sqr(n) + SumOfSquares(n - 1)
	}
}

func Difference(n int) int {
	return SquareOfSum(n) - SumOfSquares(n)
}

File: solutions/ruby/difference-of-squares/1/difference_of_squares.rb
class Squares
  def initialize(num)
    @num = num
  end

  def square_of_sum
    sum = 0
    (1..@num).each do |x|
      sum = sum + x
    end
    sum * sum
  end

  def sum_of_squares
    sum = 0
    (1..@num).each do |x|
      sum = sum + ( x * x )
    end
    sum
  end

  def difference
    square_of_sum - sum_of_squares 
  end
end

module BookKeeping
  VERSION = 4
end